k4Gen/src/components/MDIReader.cpp

```diff
@@ -0,0 +1,215 @@
+      
+      
+	//std::cout<<PHEP4<<" "<<sqrt(PHEP1*PHEP1 + PHEP2*PHEP2 + PHEP3*PHEP3)<<" "<<sqrt((PHEP1*PHEP1 + PHEP2*PHEP2 + PHEP3*PHEP3)*PHEP4*PHEP4 + PHEP5*PHEP5)<<std::endl;
+      
+	if(m_input.eof())break;
+	else if(!m_input.good())
+	  {
+	    std::cout<<"oopsie doopsie"<<std::endl;
+	    error() << "End of file reached before reading all the hits" << endmsg;
+	    return StatusCode::FAILURE;
+	  }
+	
+	IDHEP = 11;
+	CHARGE = -1;
+	if(PHEP4<0){
+	  IDHEP=-11;
+	  CHARGE=1;
+	}
+	
+	VHEP1 *=1e-9; //convert from nm to m
+	VHEP2 *=1e-9; //convert from nm to m
+	VHEP3 *=1e-9; //convert from nm to m
+	
+	//---boost section
+	PHEP4 = abs(PHEP4);
+	temp_x  = cut_z*1e-6*tan(xing) + VHEP1*sqrt(1+pow(tan(xing),2));
+	temp_px = PHEP4*tan(xing) + PHEP1*PHEP4*sqrt(1+pow(tan(xing),2));
+	temp_e  = abs( PHEP4*sqrt(1+pow(tan(xing),2)) + PHEP1*PHEP4*tan(xing) );
+	
+	VHEP1 = temp_x;
+	PHEP1 = temp_px;
+	PHEP2 = PHEP2*PHEP4;
+	PHEP3 = PHEP3*PHEP4;
+	PHEP4 = temp_e;
+	
+	//---end boost section
+	
+	VHEP1 *=1e3; //convert from m to mm
+	VHEP2 *=1e3; //convert from m to mm 
+	VHEP3 *=1e3; //convert from m to mm
+	
+	//boost to lab frame
+	
+	VHEP1=0;
+	VHEP2=0;
+	VHEP3=0;
+      }//end if guineapig
+      
+      else if(input_type=="xtrack"){
+	m_input >> VHEP3
+		>> VHEP1 >> VHEP2
+		>> PHEP1 >> PHEP2
+		>> temp_z >> PHEP4 ;
+
+	if(m_input.eof())break;
+	else if(!m_input.good())
+	  {
+	    std::cout<<"oopsie doopsie"<<std::endl;
+	    error() << "End of file reached before reading all the hits" << endmsg;
+	    return StatusCode::FAILURE;
+	  }
+	
+	//VHEP3 = VHEP3 + temp_z; //is ct necessary? prob not
+
+	//---position rotation in CLD frame
+	temp_z =  VHEP3*cos(-xing) + VHEP1*sin(-xing);
+	temp_x = -VHEP3*sin(-xing) + VHEP1*cos(-xing);
+	VHEP1 = temp_x;
+	VHEP3 = temp_z;
+
+	VHEP1 *=1e3; //convert from m to mm
+	VHEP2 *=1e3; //convert from m to mm 
+	VHEP3 *=1e3; //convert from m to mm
+
+	PHEP4 = (1.+PHEP4)*beam_energy;
+	PHEP1 = PHEP1*beam_energy;
+	PHEP2 = PHEP2*beam_energy;
+	PHEP3 = sqrt(pow(PHEP4,2) - pow(PHEP1,2) - pow(PHEP2,2) );
+
+	//---momentum rotation in CLD frame
+	temp_pz =  PHEP3*cos(-xing) + PHEP1*sin(-xing);
+	temp_px = -PHEP3*sin(-xing) + PHEP1*cos(-xing);
+	PHEP3 = temp_pz;
+	PHEP1 = temp_px;
+
+	IDHEP = 11;
+	CHARGE = -1;
+		
+      }	//end if xtrack
+
+      
+      edm4hep::MutableMCParticle particle = particles->create();
+
+      particle.setPDG(IDHEP);
+      particle.setCharge(CHARGE);
+      particle.setGeneratorStatus(ISTHEP); 
+      particle.setMomentum({
+	  (float) (PHEP1),
+	  (float) (PHEP2),
+	  (float) (PHEP3),
+	});
+      particle.setMass(PHEP5);
+      particle.setVertex({
+                          VHEP1,
+                          VHEP2,
+                          VHEP3,
+                        });
+      particle.setTime(VHEP4);
+    }
+
+    m_genphandle.put(particles);
+  return StatusCode::SUCCESS;
+}
+
+StatusCode MDIReader::finalize() 
+{
+  m_input.close();
+  std::cout<<"Fine di MDIreader"<<std::endl;
```

Comment:
Use gaudi logging instead of std::cout
```suggestion
  debug() << "Finalization of MDIREADER" << endmsg;
```

Reply:
Thanks, I changed also the other `std::cout` to `debug()`